srv/salt/_modules/cephdisks.py

```diff
@@ -1,612 +1,130 @@
-    """
-    if match:
-        log.warning("Using argument {} for match".format(match))
-    else:
-        match = _seek('ceph:modules:cephdisks:device:match'.split(':'), __pillar__)
-        if match:
-            log.warning("Using pillar value {} for match".format(match))
-        else:
-            match = "-name ata* -o -name scsi* -o -name nvme*"
-    return match
 
+def get_(disk_path) -> dict:
+    import pdb;pdb.set_trace()
+    return Device(disk_path).json_report()
 
-def _pathname_setting(pathname):
-    """
-    Predence is command line, pillar, then default
-    """
-    if pathname:
-        log.warning("Using argument {} for pathname".format(pathname))
-    else:
-        pathname = _seek('ceph:modules:cephdisks:device:pathname'.split(':'), __pillar__)
-        if pathname:
-            log.warning("Using pillar value {} for pathname".format(pathname))
-        else:
-            pathname = "/dev/disk/by-id"
-    return pathname
 
+def find_by_osd_id(osd_id, **kwargs):
+    return [
+        x.json_report() for x in Inventory(**kwargs).find_by_osd_id(osd_id)
+    ]
 
-def _seek(keys, saltdict):
-    """
-    Recursively check for nested keys
-    """
-    if keys and keys[0] in saltdict:
-        if keys[1:]:
-            return _seek(keys[1:], saltdict[keys[0]])
-        return saltdict[keys[0]]
-    return None
 
+def devices(**kwargs):
+    return [x.path for x in Inventory(**kwargs)._filter() if x.path]
 
-def _prefer_underscores(devicenames):
-    """
-    Many symlinks in /dev/disk/by-id refer to the same device.  The
-    most descriptive names have the most underscores.  These are likely
-    the most useful to the admin.
 
-    In the worst case, return the last device
-    """
-    index = -1
-    count = 0
-    for _idx, device in enumerate(devicenames):
-        underscores = device.count('_')
-        if underscores > count:
-            count = underscores
-            index = _idx
-    return index
+def all_(**kwargs):
+    return [x.json_report() for x in Inventory(**kwargs).devices.devices]
 
 
 def list_(**kwargs):
-    """
-    List the disks
-    """
-    hwd = HardwareDetections(**kwargs)
-    return hwd.assemble_device_list()
+    kwargs.update(dict(used_by_ceph=True))
+    return [x.json_report() for x in Inventory(**kwargs)._filter()]
 
 
-def filter_(key="Device File", **kwargs):
-    """
-    Return list of specified key
-    """
-    hwd = HardwareDetections(**kwargs)
-    result = hwd.assemble_device_list()
-    results = [device[key] for device in result]
-    return sorted(results)
+def osd_list(**kwargs):
+    return Inventory(**kwargs).osd_list()
 
 
-def version():
-    """
-    Displays version
-    """
-    return VERSION
+def help_():
+    print("HELP DUMMY")
 
 
 __func_alias__ = {
-                'list_': 'list',
-                'device_': 'device',
-                'filter_': 'filter',
-                 }
+    'all_': 'all',
+    'list_': 'list',
```

Comment:
`cephdisks.list` would previously return all disks. This changes it so that you now have to use `cephdisks.all` to get all disks, and `cephdisks.list` only if you want ones already used by ceph. srv/modules/runners/mgr_orch.py relied on the old behaviour, but that's OK, we can change that to use cephdisks.all with no trouble (and that runner requires some other tweaks to deal with ceph-volume output, which I'll take care of). Do we have to make similar changes elsewhere where cephdisks.list is currently used?

Reply:
It's still a WIP branch and we can change the func names at all times.
Defining terminology is tough again :/

maybe:

`cephdisks.used`

for disks that are used by ceph

`cephdisks.all`

for all disks regardless of used or not

Reply:
> . Do we have to make similar changes elsewhere where cephdisks.list is currently used?

I was looking at the necessary changes the other day.. I can probably fix them within this PR. Otherwise I'll let people know, mhk?

Reply:
Sounds good.